frontend/javascripts/oxalis/model/helpers/nml_helpers.ts

```diff
@@ -517,32 +542,66 @@ export class NmlParseError extends Error {
   name = "NmlParseError";
 }
 
-function _parseInt(obj: Record<string, string>, key: string, defaultValue?: number): number {
+function _parseInt<T>(
+  obj: Record<string, string>,
+  key: string,
+  options?: {
+    defaultValue: number | T;
+  },
+): number | T {
   if (obj[key] == null || obj[key].length === 0) {
-    if (defaultValue == null) {
+    if (options == null) {
       throw new NmlParseError(`${messages["nml.expected_attribute_missing"]} ${key}`);
     } else {
-      return defaultValue;
+      return options.defaultValue;
     }
   }
 
   return Number.parseInt(obj[key], 10);
 }
 
-function _parseFloat(obj: Record<string, string>, key: string, defaultValue?: number): number {
+function _parseFloat<T>(
+  obj: Record<string, string>,
+  key: string,
+  options?: {
+    defaultValue: number | T;
+  },
+): number | T {
   if (obj[key] == null || obj[key].length === 0) {
-    if (defaultValue == null) {
+    if (options == null) {
       throw new NmlParseError(`${messages["nml.expected_attribute_missing"]} ${key}`);
     } else {
-      return defaultValue;
+      return options.defaultValue;
     }
   }
 
   return Number.parseFloat(obj[key]);
 }
 
+function _parseStringArray(
+  obj: Record<string, string>,
+  prefix: string,
+  options?: {
+    defaultValue: string[] | undefined;
+  },
+): string[] | undefined {
+  const indices = Object.keys(obj)
+    .map((key) => key.split(`${prefix}-`))
+    .filter((splitElements) => splitElements.length === 2)
```

Comment:
Object.keys returns all xml properties of the current tag. so, `treeId` would also be split using the prefix. the split array would only contain one entry. we need to filter this away, because we are not interested in those.